src/api/spec/features/webui/packages_spec.rb

```diff
@@ -25,14 +27,117 @@
 
     scenario "from another user's project" do
       login user
-      visit package_show_path(project: other_user.home_project, package: package)
+      visit package_show_path(project: other_user.home_project, package: branch_package)
 
       click_link("Branch package")
       click_button("Ok")
 
       expect(page).to have_text("Successfully branched package")
       expect(page.current_path).to eq(
-        package_show_path(project: user.branch_project_name(other_user.home_project_name), package: package))
+        package_show_path(project: user.branch_project_name(other_user.home_project_name), package: branch_package))
+    end
+  end
+
+  describe "editing package files" do
+    let(:file_edit_test_package) { create(:package_with_file, name: "file_edit_test_package", project: user.home_project) }
+
+    before do
+      login(user)
+      visit package_show_path(project: user.home_project, package: file_edit_test_package)
+    end
+
+    scenario "editing an existing file" do
+      # somefile.txt is a file of our test package
+      click_link("somefile.txt")
+      # Workaround to update codemirror text field
+      execute_script("$('.CodeMirror')[0].CodeMirror.setValue('added some new text')")
+      click_button("Save")
+
+      expect(page).to have_text("The file 'somefile.txt' has been successfully saved.")
+      expect(file_edit_test_package.source_file("somefile.txt")).to eq("added some new text")
+    end
+  end
+
+  describe "adding files to package" do
+    let(:files_test_package) { create(:package_with_file, name: "files_test_package", project: user.home_project) }
+    let(:file_to_upload) { File.expand_path(Rails.root.join("test/texts/chinese.txt")) }
+
+    before do
+      login(user)
+      visit package_show_path(project: user.home_project, package: files_test_package)
+      click_link("Add file")
+    end
+
+    scenario "adding a file that doesn't exist yet" do
```

Comment:
If you mean that single test: I wouldn't split it from the others, since they are all about the same workflow.

If you mean the whole "adding files" section: If we are okay to not test the view, I can move it there.